showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java

```diff
@@ -0,0 +1,183 @@
+   *
+   * The provider's credentials do not get altered when adding audience.
+   * They are recreated and used only inside `ClientContext`
+   * @throws IOException
+   * @throws URISyntaxException
+   */
+  @Test
+  public void testClientWithGdchCredentialWithValidAudience_correct()
+      throws IOException, URISyntaxException {
+    String audience = "valid-audience";
+    settings = settings.toBuilder().setGdchApiAudience(audience).build();
+    Exception unexpected = getExceptionFromClientCreation();
+    assertNull(unexpected);
+    GdchCredentials fromClient =
+            (GdchCredentials) client.getSettings().getCredentialsProvider().getCredentials();
+
+    assertSame(credentials, fromClient);
+  }
+
+  /**
+   * Confirms that the context recreates and contains a different credential when `gdchApiAudience` is specified
+   * @throws IOException
+   * @throws URISyntaxException
+   */
+  @Test
+  public void testClientWithGdchCredentialWithValidAudience_modifiesContextCredentials()
+      throws IOException, URISyntaxException {
+    String audience = "valid-audience";
+    ClientContext context = ClientContext.newBuilder()
+            .setCredentials(credentials)
+            .build();
+    settings = settings.newBuilder(context).build();
+    Exception unexpected = getExceptionFromClientCreation();
+    assertNull(unexpected);
+    GdchCredentials fromClient =
+            (GdchCredentials) client.getSettings().getCredentialsProvider().getCredentials();
+    URI audienceFromClientProvider = fromClient.getApiAudience();
+    assertNotNull(audienceFromClientProvider);
+    assertTrue(audienceFromClientProvider.equals(new URI(audience)));
+    assertTrue(
+            "GDCH credentials with audience should be GdchCredentials",
+            credentials instanceof GdchCredentials);
+    GdchCredentials fromContext = null;
+    assertNotSame(fromContext, );
+  }
+
+  @Test
+  public void testClientWithGdchCredentialWithInvalidAudience_throws() throws IOException {
+    settings = settings.toBuilder().setGdchApiAudience("$invalid-audience:").build();
+    Exception expected = getExceptionFromClientCreation();
+    assertNotNull(expected);
+    assertTrue(expected instanceof IllegalArgumentException);
+    assertTrue(expected.getMessage().contains("audience string is not a valid URI"));
+  }
+
+  @Test
+  public void testClientWithNonGdchCredentialWithAnyAudience_throws() throws IOException {
+    settings =
+        settings
+            .toBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setGdchApiAudience("any-audience")
+            .build();
+    Exception expected = getExceptionFromClientCreation();
+    assertNotNull(expected);
+    assertTrue(expected instanceof IllegalArgumentException);
+    assertTrue(
+        expected.getMessage().contains("audience can only be set when using GdchCredentials"));
+  }
+
+  private Exception getExceptionFromClientCreation() {
```

Comment:
This helper method is actually more confusing to me. Since most of the cases are not throwing, I don't think we need this and we can `assertThrows` for cases that need it, which is more readable as well.

Reply:
`assertThrows` indeed seems more convenient. Using it now!